File: client/src/modules/Battle/Logic/Weapons/Bomb/BombWeapon.js
var BombWeapon = Weapon.extend({

    ctor: function(player, pos, mapMgr, mapGUI, cellSize) {
        this._super(player, pos, mapMgr, mapGUI, cellSize, "res/noItem.png");

        this.numUse = WEAPONS.TIME_USE_BOMB
    },

    getTimeUseWeapon: function () {
        return this.numUse
    },

    attack: function () {
        let posLogic = this.mapGUI.convertIndexToPosLogic(this.player)
        if (this.checkCondition(posLogic)) {
            let bomb = new Bomb(this.player, this.mapGUI, posLogic, this.cellSize)
            bomb.setScale(
                this.cellSize / bomb.width,
                this.cellSize / bomb.height
            )
            bomb.setPosition(this.mapGUI.convertIndexToPosUI(posLogic.x, posLogic.y))
            this.mapGUI.listBomb.push(bomb)
            this.mapGUI.addChild(bomb)
        }
    },

    checkCondition: function (posLogic) {
        // Check trong o co bomb khong?
        if (this.mapGUI.isBomb(posLogic.x, posLogic.y)) {
            return false
        }

        // Check player con dat duoc bom khong?
        if  (!this.player.canBomb()) {
            return false
        }
        return true
    }
});

File: server/src/main/java/battle/model/weapons/bomb/BombWeapon.java
package battle.model.weapons.bomb;

import battle.BattleMgr;
import battle.MapLogic;
import battle.model.Character;
import battle.model.Coordinate;
import battle.model.weapons.Weapon;
import config.battle.WeaponsConfig;

public class BombWeapon extends Weapon {
    public BombWeapon(Character character, Coordinate currentPos, MapLogic mapMgr, BattleMgr battleMgr, double cellSize) {
        super(character, currentPos, mapMgr, battleMgr, cellSize);
        this.numUse = WeaponsConfig.TIME_USE_BOMB;
    }

    public int getTimeUseWeapon() {
        return this.numUse;
    }

    public void attack() {
        Coordinate posLogic = this.mapGUI.convertIndexToPosLogic(this.player.currentPos);
        if (this.checkCondition(posLogic)) {
            Bomb bomb = new Bomb(this.player, this.mapGUI, posLogic, this.cellSize);
            this.mapGUI.listBomb.add(bomb);
        }
    }

    public boolean checkCondition(Coordinate posLogic) {
        // Check trong o co bomb khong?
        if (this.mapGUI.isBomb((int) posLogic.getX(), (int) posLogic.getY())) {
            return false;
        }

        // Check player con dat duoc bom khong?
        return this.player.canBomb();
    }
}
